### PR TITLE
Add support for env variable with DD prefixes

### PR DESCRIPTION
**Pull request by skarimo**
Closes: #402 

This PR adds support for: DD_API_KEY, DD_APP_KEY and DD_HOST env variables. Retained support for DATADOG_API_KEY, DATADOG_APP_KEY and DATADOG_HOST for backward compatibility

### CODE REVIEW

File: datadog/provider_test.go [hunk 1]
```diff
@@ -162,10 +162,10 @@ func testAccPreCheck(t *testing.T) {
 	if isReplaying() {
 		return
 	}
-	if v := os.Getenv("DATADOG_API_KEY"); v == "" {
-		t.Fatal("DATADOG_API_KEY must be set for acceptance tests")
+	if v := os.Getenv("DD_API_KEY"); v == "" {
```
Comment (nmuesch) [inline datadog/provider_test.go:165]: I believe the CI relies on the secrets being this name. Can we do an `or` and look for the old one as well for now too?

File: datadog/provider.go [hunk 2]
```diff
@@ -23,17 +23,17 @@ func Provider() terraform.ResourceProvider {
 			"api_key": {
 				Type:        schema.TypeString,
 				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("DATADOG_API_KEY", nil),
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{"DD_API_KEY", "DATADOG_API_KEY"}, nil),
```
Comment (nmuesch) [inline datadog/provider.go:26]: Nice, TIL `MultiEnvDefaultFunc` I'm torn on which variable should be first here. According to https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L287, the first available variable in the list that has a value is used.

I'm wondering if the safest path is to have the old value (`DATADOG_`) be the first one, so in the case a user has two set, i.e. one for testing and another for another org, we'd choose the old value. WDYT?

Comment (skarimo) [reply datadog/provider.go]: Definitely agree with you there, just in case customer has both set previously.

File: datadog/provider_test.go (hunk 1 above)
Comment (skarimo) [reply datadog/provider_test.go]: Broke it down into its own functions. Let me know what you think:

```
func testAccPreCheck(t *testing.T) {
	if isReplaying() {
		return
	}
	if !isAPIKeySet() {
		t.Fatal("DD_API_KEY must be set for acceptance tests")
	}
	if !isAPPKeySet() {
		t.Fatal("DD_APP_KEY must be set for acceptance tests")
	}
}
```

**Pull request merged**